src/catalog/__tests__/catalogEntities.spec.js

```
@@ -0,0 +1,259 @@
+      apiKinds.PROXY,
+      apiKinds.PUBLIC_KEY,
+      apiKinds.CLUSTER,
+    ].forEach((apiKind) => {
+      const entity = {
+        metadata: {
+          cloudUrl: 'https://test-cloud-url.com',
+          kind: apiKind,
+          name: 'test-entity-name',
+          namespace: 'test-namespace',
+        },
+      };
+
+      it(`returns url for |${apiKind}| entity`, () => {
+        const url = `${entity.metadata.cloudUrl}/projects/${entity.metadata.namespace}`;
+
+        switch (entity.metadata.kind) {
+          case apiKinds.RHEL_LICENSE:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/rhel?name=${entity.metadata.name}&details`
+            );
+            break;
+          case apiKinds.PROXY:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/proxies?name=${entity.metadata.name}&details`
+            );
+            break;
+          case apiKinds.PUBLIC_KEY:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/keypairs?name=${entity.metadata.name}&details`
+            );
+            break;
+          case apiKinds.CLUSTER:
+            expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+              `${url}/clusters/${entity.metadata.name}`
+            );
+            break;
+        }
+      });
+    });
+
+    Object.values(apiCredentialKinds).forEach((apiKind) => {
+      const entity = {
+        metadata: {
+          cloudUrl: 'https://test-cloud-url.com',
+          kind: apiKind,
+          name: 'test-entity-name',
+          namespace: 'test-namespace',
+        },
+      };
+
+      it(`returns url for |${apiKind}| entity`, () => {
+        const url = `${entity.metadata.cloudUrl}/projects/${entity.metadata.namespace}`;
+
+        expect(catalogEntities.generateEntityUrl(entity)).toEqual(
+          `${url}/creds?name=${entity.metadata.name}&details`
+        );
+      });
+    });
```

Looks like you could combine all this into a single loop with

```
    [
      apiKinds.RHEL_LICENSE,
      apiKinds.PROXY,
      apiKinds.PUBLIC_KEY,
      apiKinds.CLUSTER,
      ...Object.values(apiCredentialKinds),
    ].forEach((apiKind) => {
```

And then use the default case to test `if (Object.values(apiCredentialKinds).includes(apiKind))` and if not, throw an error (instead of silently doing nothing) about an unexpected kind we don't have a test for yet.

---

Makes sense, fixed